revolutionpi/io_util.go

```diff
@@ -1,57 +1,64 @@
-	KB_STOP_IO                = ioctlAddress(22) // stop/start IO communication, can be used for I/O simulation
-	KB_CONFIG_STOP            = ioctlAddress(23) // for download of configuration to Master Gateway: stop IO communication completely
-	KB_CONFIG_SEND            = ioctlAddress(24) // for download of configuration to Master Gateway: download config data
-	KB_CONFIG_START           = ioctlAddress(25) // for download of configuration to Master Gateway: restart IO communication
-	KB_SET_OUTPUT_WATCHDOG    = ioctlAddress(26) // activate a watchdog for this handle. If write is not called for a given period all outputs are set to 0
-	KB_SET_POS                = ioctlAddress(27) // set the f_pos, the unsigned int * is used to interpret the pos value
-	KB_AIO_CALIBRATE          = ioctlAddress(28)
+	kbCmd1                 = ioctlAddress(10) // for test only
+	kbCmd2                 = ioctlAddress(11) // for test only
+	kbReset                = ioctlAddress(12) // reset the piControl driver including the config file
+	kbGetDeviceInfoList    = ioctlAddress(13) // get the device info of all detected devices
+	kbGetDeviceInfo        = ioctlAddress(14) // get the device info of one device
+	kbGetValue             = ioctlAddress(15) // get the value of one bit in the process image
+	kbSetValue             = ioctlAddress(16) // set the value of one bit in the process image
+	kbFindVariable         = ioctlAddress(17) // find a variable defined in piCtory
+	kbSetExportedOutputs   = ioctlAddress(18) // copy the exported outputs from a application process image to the real process image
+	kbUpdateDeviceFirmware = ioctlAddress(19) // try to update the firmware of connected devices
+	kbDIOResetCounter      = ioctlAddress(20) // set a counter or encoder to 0
+	kbGetLastMessage       = ioctlAddress(21) // copy the last error message
+	kbStopIO               = ioctlAddress(22) // stop/start IO communication, can be used for I/O simulation
+	kbConfigStop           = ioctlAddress(23) // for download of configuration to Master Gateway: stop IO communication completely
+	kbConfigSend           = ioctlAddress(24) // for download of configuration to Master Gateway: download config data
+	kbConfigStart          = ioctlAddress(25) // for download of configuration to Master Gateway: restart IO communication
+	// activate a watchdog for this handle. If write is not called for a given period all outputs are set to 0.
+	kbSetOutputWatchdog = ioctlAddress(26)
+	kbSetPos            = ioctlAddress(27) // set the f_pos, the unsigned int * is used to interpret the pos value
+	kbAIOCalibrate      = ioctlAddress(28)
+	kbWaitForEvent      = ioctlAddress(50) // wait for an event. This call is normally blocking
 )
 
-var KB_WAIT_FOR_EVENT = ioctlAddress(50) // wait for an event. This call is normally blocking
-
+// SPIVariable is the struct representing an address when reading from the board.
+// use kbFindVariable with ioctl to populate the struct.
```

Comment:
:+1: I also really like the comments you've left here about which constants above get used where down here. Thanks!